src/Promitor.Core.Scraping/Configuration/Serialization/v1/Core/AzureMetadataDeserializer.cs

```diff
@@ -1,32 +1,28 @@
-﻿using GuardNet;
-using Microsoft.Extensions.Logging;
+﻿using Microsoft.Extensions.Logging;
 using Promitor.Core.Scraping.Configuration.Serialization.v1.Model;
 using YamlDotNet.RepresentationModel;
 
 namespace Promitor.Core.Scraping.Configuration.Serialization.v1.Core
 {
-    internal class AzureMetadataDeserializer : Deserializer<AzureMetadataV1>
+    public class AzureMetadataDeserializer : Deserializer<AzureMetadataV1>
     {
-        internal AzureMetadataDeserializer(ILogger logger) : base(logger)
+        private const string TenantIdTag = "tenantId";
+        private const string SubscriptionIdTag = "subscriptionId";
+        private const string ResourceGroupNameTag = "resourceGroupName";
+
+        public AzureMetadataDeserializer(ILogger logger) : base(logger)
         {
         }
 
-        internal override AzureMetadataV1 Deserialize(YamlMappingNode node)
+        public override AzureMetadataV1 Deserialize(YamlMappingNode node)
         {
-            Guard.NotNull(node, nameof(node));
-
-            var tenantId = node.Children[new YamlScalarNode("tenantId")];
-            var subscriptionId = node.Children[new YamlScalarNode("subscriptionId")];
-            var resourceGroupName = node.Children[new YamlScalarNode("resourceGroupName")];
-
-            var azureMetadata = new AzureMetadataV1
-            {
-                TenantId = tenantId?.ToString(),
-                SubscriptionId = subscriptionId?.ToString(),
-                ResourceGroupName = resourceGroupName?.ToString()
-            };
+            var metadata = new AzureMetadataV1();
 
-            return azureMetadata;
+            metadata.TenantId = GetString(node, TenantIdTag);
```

Comment:
I'll convert them to extension methods and add tests. I couldn't really add tests for these previously because it's a base class, and the tests are against the implementations, but there's a test for every field in all of the deserializers, so there's effectively a test for all of these (since I added a test before implementing each of these methods).

Reply:
That would be great, thanks!

Test are optional if you are up for it, but don't feel obliged to do it as we indirectly do it indeed. Will create an issue then to create them "some day" but is less critical and not blocking here

Reply:
Ok - I've created a new class called `YamlMappingNodeExtensions` and converted all of these to extension methods. I've also added tests for them all - it's easy enough to do, and better to do it now. In my experience, "some day" quite often never happens ;-)

Reply:
That's about right 😅 Thanks!